### PR TITLE
Add default value for wildfly, format and rework RegexPattern

### LINKED ISSUE #114: Default config doesn't work/exist
Since Title and Commit checks are implicitly enabled, I shouldn't have to set any checks at all if I want to use the default config. However

```
wildfly:
```
doesn't work, same as 

```
wildfly:
 format:
```
or even empty config

### PR DESCRIPTION
**Pull request by The-Huginn**
Resolves #114 and resolves a bug found by @petrberan , when only `RegexPattern.enabled` property is set to `true` and `RegexPattern.message` then results in `null`.

### CODE REVIEW

File: src/test/java/io/xstefank/wildfly/bot/PREditTest.java [hunk 1]
```diff
@@ -38,14 +38,17 @@ void setUp() {
 
     @Test
     void configFileNullTest() throws IOException {
```
Comment (petrberan) [inline src/test/java/io/xstefank/wildfly/bot/PREditTest.java:40]: This test can be deleted once the #118 is merged

File: src/test/java/io/xstefank/wildfly/bot/PRFormatOverrideMessage.java [hunk 2]
```diff
@@ -95,4 +95,76 @@ public void testOverridingCommitMessage() throws IOException {
                             .formatted(String.format("- For commit: \"%s\" (%s) - Custom commit message", commitMessage, gitHubJson.commitSHA())));
                 });
     }
+
+    @Test
+    public void testOverridingTitleEnabledToTrue() throws IOException {
```
Comment (petrberan) [inline src/test/java/io/xstefank/wildfly/bot/PRFormatOverrideMessage.java:100]: This can use the JSON Builder for all the constants. I'd like to believe the Builder was implemented during the creation of this PR. 

It would also benefit from using the Utils method once #118 is merged. New place for the test should be PRCommitCheckTest

File: src/test/java/io/xstefank/wildfly/bot/PRTitleCheckTest.java [hunk 3]
```diff
@@ -61,6 +65,51 @@ void incorrectTitleCheckFailTest() throws IOException {
             });
     }
 
+    @Test
```
Comment (petrberan) [inline src/test/java/io/xstefank/wildfly/bot/PRTitleCheckTest.java:68]: Same as above, especially the builder.

File: src/test/java/io/xstefank/wildfly/bot/PREditTest.java (hunk 1 above)
Comment (The-Huginn) [reply src/test/java/io/xstefank/wildfly/bot/PREditTest.java:40]: I agree, I just wanted to use multi line string instead of previous concatenation.

File: src/test/java/io/xstefank/wildfly/bot/PRFormatOverrideMessage.java (hunk 2 above)
Comment (The-Huginn) [reply src/test/java/io/xstefank/wildfly/bot/PRFormatOverrideMessage.java:100]: The builder is not currently in any PR. I am working on different issues and I have created the builder there. Soon I will open PR with it. It will however add new features, so it might get delayed due to that.

**Comment by The-Huginn** [comment]
Please review @petrberan @xstefank

File: src/test/java/io/xstefank/wildfly/bot/PRTitleCheckTest.java [hunk 4]
```diff
@@ -61,6 +61,47 @@ void incorrectTitleCheckFailTest() throws IOException {
...
+    void incorrectTitleCheckFailFormatConfigMissingTest() throws IOException {
```
Comment (petrberan) [inline src/test/java/io/xstefank/wildfly/bot/PRTitleCheckTest.java:65]: I'm missing positive test cases, aka `correctTitleCheckSuccessFormatConfigMissingTest` and `correctTitleCheckSuccessFormatConfigNullTest` when `gitHubJson = GitHubJson.builder(VALID_PR_TEMPLATE_JSON).build()`. Given that they should pass of course.

**Comment by petrberan** [comment]
@The-Huginn What is the behavior when the config is

```
wildfly:
  format:
    title:
``` 
?

**Comment by The-Huginn** [comment]
Thanks @petrberan . I have added 4 more tests. 2 are testing the suggested correct behaviors from your [review](https://github.com/xstefank/wildfly-github-bot/pull/116#discussion_r1289730147) and 2 more for the [emptry title entry](https://github.com/xstefank/wildfly-github-bot/pull/116#issuecomment-1672758357)

**Comment by The-Huginn** [comment]
Also I can add tests for empty commit and description entries, as I have updated them too.

**Comment by petrberan** [comment]
Please do @The-Huginn

File: src/main/java/io/xstefank/wildfly/bot/TriagePullRequestProcessor.java [hunk 5]
```diff
@@ -35,15 +35,13 @@ void onPullRequestOpened(@PullRequest.Opened GHEventPayload.PullRequest pullRequ
         GHPullRequest pullRequest = pullRequestPayload.getPullRequest();
         List<String> mentions = new ArrayList<>();
 
-        if (wildflyBotConfigFile.wildfly.rules != null) {
-            for (WildFlyRule rule : wildflyBotConfigFile.wildfly.rules) {
-                if (Matcher.matches(pullRequest, rule)) {
-                    LOG.infof("Pull Request %s was matched with a rule with the id: %s.", pullRequest.getTitle(),
+        for (WildFlyRule rule : wildflyBotConfigFile.wildfly.rules) {
+            if (Matcher.matches(pullRequest, rule)) {
+                LOG.debugf("Pull Request %s was matched with a rule with the id: %s.", pullRequest.getTitle(),
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/TriagePullRequestProcessor.java:40]: this is now infof, so I think you need to rebase

**Review by xstefank — CHANGES_REQUESTED** [review]

**Review by xstefank — CHANGES_REQUESTED** [review]

**Review by xstefank — APPROVED** [review]
